projects/plugins/social/src/js/components/panel/description.jsx

```diff
@@ -0,0 +1,55 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+const descriptions = {
+	start: __(
+		'Start sharing your posts by connecting your social media accounts.',
+		'jetpack-social'
+	),
+	enabled: __(
+		'This post will be shared on all your enabled social media accounts the moment you publish the post.',
+		'jetpack-social'
+	),
+	disabled: __(
+		'Use this tool to share your post on all your social media accounts.',
+		'jetpack-social'
+	),
+	published: __( 'Posts can only be shared as they are first published.', 'jetpack-social' ),
+};
+
+const getDescription = ( {
+	isPostPublished,
+	isPublicizeEnabled,
+	hasConnections,
+	hasEnabledConnections,
+} ) => {
+	if ( ! hasConnections ) {
+		return descriptions.start;
+	}
+	if ( isPostPublished ) {
+		return descriptions.published;
+	}
+	if ( isPublicizeEnabled && hasEnabledConnections ) {
+		return descriptions.enabled;
+	}
+
+	return descriptions.disabled;
+};
+
+/**
+ * A component that renders a contextual description for
+ * the Publicize publishing panel.
+ *
+ * @param {object} props                        - The component properties.
+ * @param {boolean} props.isPostPublished       - Whether the post has been published.
+ * @param {boolean} props.isPublicizeEnabled    - Whether the main publicize toggle is enabled.
+ * @param {boolean} props.hasConnections        - Whether we have any Publicize connections.
+ * @param {boolean} props.hasEnabledConnections - Whether any connections are enabled.
+ * @returns {object} The description component.
+ */
+export default function Description( props ) {
+	const descrip = getDescription( props );
+	return <div>{ descrip }</div>;
```

Comment:
```suggestion
	return <div>{ getDescription( props ) }</div>;
```
For some reason I really don't like the `descrip` abbreviation 😆

Reply:
The entire function could probably be simplified to this if we wanted to:

```js
const Description = props => <div>{ getDescription( props ) }</div>;

export default Description;
```

Reply:
Yes good point, I think this is hangover from when I was doing more in that component. I'll get that changed.

Reply:
Addressed in ac1357a